### PR TITLE
R specs

### PR DESCRIPTION
**Pull request by PromzzyKoncepts**
# Hey There
> In this PR, we met the project requirements

### Project requirements
- Add unit tests for all the methods in all the classes:
  - `Book`
  - `Person`
  - `Student`
  - `Teacher`
  - `Classroom`
  - `Rental`

## KIndly Review

### CODE REVIEW

File: spec/classroom_spec.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require_relative '../classroom'
```
Comment (HAMMAS-SALEEM) [inline spec/classroom_spec.rb:1]: image

- [ ] You didn't test **add_students** method shown in the image above :arrow_up: Kindly test this test to follow the requirements and fully test your classes :+1:

**Hint: You can see the below test to get an idea :arrow_down:**

```ruby
    it 'expect the classroom to have students' do
      student = Student.new(23, 'Maxwell', true)
      classroom.adding_students(student)
      expect(classroom.students).to eq [student]
    end
```

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    it 'expects Margaret age to be 45' do
+      expect(person.age).to eq 45
+    end
+    it 'expects Person.name to be Margaret' do
+      expect(person.name).to eq 'Margaret'
+    end
+
+    it 'expect can_use_services to equal true' do
+      expect(person.can_use_services?).to eq true
+    end
```
Comment (HAMMAS-SALEEM) [inline spec/person_spec.rb]: - [ ] When you create a person than a rental array is created :thinking: You didn't check that Kindly test your rentals array using the following code :arrow_down:

`expect(person.rentals).to eql []`

File: spec/person_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  context 'containing a list of names and ages' do
+    person = Person.new(45, 'Margaret')
+
+    it 'expects Margaret age to be 45' do
+      expect(person.age).to eq 45
+    end
+    it 'expects Person.name to be Margaret' do
+      expect(person.name).to eq 'Margaret'
+    end
+
```
Comment (HAMMAS-SALEEM) [inline spec/person_spec.rb:14]: image

- [ ] You didn't test the correct_name method shown in the image above :thinking: Kindly test it :+1: to fully test your person class :blush:

File: spec/person_spec.rb (hunk 2 above)
Comment (HAMMAS-SALEEM) [inline spec/person_spec.rb:17]: image

- [ ] You have tested this method :clap: but you didn't cover all the scenarios :fearful: means what if

1. The person doesn't have an age more than or equal to 18
2. The person doesn't have parent_permission true
3. The person doesn't have an age more than or equal to 18 and doesn't have parent_permission true

Kindly cover all the scenarios to fully test your classes with all the exceptional or expected cases :+1:

File: spec/student_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+require_relative '../students'
+require_relative '../persons'
+
+describe Student do
+  context 'with an input of names and ages' do
```
Comment (HAMMAS-SALEEM) [inline spec/student_spec.rb:5]: image

- [ ] You didn't test **play_hooky** method :fearful: Kindly test this method to test your student class fully :+1:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @PromzzyKoncepts, @belalgamal46 :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**